apps/web/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { useState } from 'react';
+import './SearchBar.scss';
+
+export interface Props {
+  /** What happens when we use a searchstring to search for results? */
+  handleSearch: (searchString: string) => void;
+  /**  An optional custom test ID for the component.  */
+  customTestId?: string;
+}
+
+// need to add jsdoc comment
+const SearchBar = ({
+  handleSearch,
+  customTestId = 'searchbar',
+}: Props): React.ReactElement => {
+  /** Tracks the string value within input */
+  const [searchString, setSearchString] = useState<string>('');
+
+  /**
+   * Handles 'Enter' key press & invokes `handleSearch` func
+   *
+   * @param {React.KeyboardEvent} event - The keyboard event that triggered the function.
+   */
+  const handleKeyDown = (event: React.KeyboardEvent): void => {
```

Comment:
Is the `KeyboardEvent` the only event that can trigger search? 

What about a click event?

Reply:
As for the `searchIcon` & `closeIcon` I have added and positioned them both, using ::before & ::after. After doing so, I was looking up how to add an onClick to clear the input & trigger handleSearch, but I haven't been able to find much besides this workaround on stack-overflow. Wanted to see what you thought.

#TLDR
"This is not possible; pseudo-elements are not part of the DOM at all so you can't bind any events directly to them, you can only bind to their parent elements.

If you must have a click handler on the red region only, you have to make a child element, like a span, place it right after the opening <p> tag, apply styles to p span instead of p:before, and bind to it."

https://stackoverflow.com/questions/7478336/only-detect-click-event-on-pseudo-element

Reply:
I need to figure out how to get the `closeButton` to be hidden when the input is empty, and how to hide it when the `closeButton` is pressed & the input is cleared. I tried conditionally rendering the `span` when the ref is `null` equal to an empty string but it did not work.